microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/core/java/validators/annotations/AnnotationRulesJavaASTValidator.java

```diff
@@ -99,9 +102,17 @@ private void validateAnnotationAttributeValue(Expression attributeValueExpr, Ann
 
 		// Get the value of the AST attribute
 		String valueAsString = attributeValueExpr.toString();
-		if (StringUtils.isEmpty(valueAsString)) {
+
+		// Ensure value of AST attribute is a valid integer or an expression that can be evaluated
+		if ((StringUtils.isEmpty(valueAsString)) || (!(attributeValueExpr instanceof NumberLiteral)
+				&& !(attributeValueExpr instanceof PrefixExpression && isSignedInteger(attributeValueExpr))
+				&& !(attributeValueExpr instanceof InfixExpression && isInfixIntegerExpression(attributeValueExpr)))) {

```

Comment:
Could you also move the `attributeValueExpr instanceof ...` within the corresponding helper method ? Given that the methods take just an `Expression`, it should work and keep things even simpler.